runtime/src/rpc.rs

```diff
@@ -232,39 +232,30 @@ impl InterBtcParachain {
 
         *next_nonce = ret.saturating_add(1);
 
-        ret
+        Ok(ret)
     }
 
-    async fn query_finalized<Address>(
-        &self,
-        address: Address,
-    ) -> Result<Option<<Address::Target as DecodeWithMetadata>::Target>, Error>
+    async fn query_finalized<Address>(&self, address: Address) -> Result<Option<Address::Target>, Error>
     where
         Address: StorageAddress<IsFetchable = Yes>,
     {
         let hash = self.get_finalized_block_hash().await?;
-        Ok(self.api.storage().fetch(&address, hash).await?)
+        Ok(self.api.storage().at(hash).fetch(&address).await?)
     }
 
-    async fn query_finalized_or_error<Address>(
-        &self,
-        address: Address,
-    ) -> Result<<Address::Target as DecodeWithMetadata>::Target, Error>
+    async fn query_finalized_or_error<Address>(&self, address: Address) -> Result<Address::Target, Error>
     where
         Address: StorageAddress<IsFetchable = Yes>,
     {
         self.query_finalized(address).await?.ok_or(Error::StorageItemNotFound)
     }
 
-    async fn query_finalized_or_default<Address>(
-        &self,
-        address: Address,
-    ) -> Result<<Address::Target as DecodeWithMetadata>::Target, Error>
+    async fn query_finalized_or_default<Address>(&self, address: Address) -> Result<Option<Address::Target>, Error>
     where
         Address: StorageAddress<IsFetchable = Yes, IsDefaultable = Yes>,
     {
         let hash = self.get_finalized_block_hash().await?;
-        Ok(self.api.storage().fetch_or_default(&address, hash).await?)
+        Ok(Some(self.api.storage().at(hash).fetch_or_default(&address).await?))
```

Comment:
Why wrap in option? Looks like you never return none?

Reply:
removed Option